src/ssl.c

```diff
@@ -10181,23 +10182,65 @@ void* wolfSSL_X509_get_ext_d2i(const WOLFSSL_X509* x509, int nid, int* c,
         }
 
         case CRL_DIST_OID:
+
             if (x509->CRLdistSet && x509->CRLInfo != NULL) {
                 if (c != NULL) {
                     *c = x509->CRLdistCrit;
                 }
-                obj = wolfSSL_ASN1_OBJECT_new();
-                if (obj == NULL) {
-                    WOLFSSL_MSG("Issue creating WOLFSSL_ASN1_OBJECT struct");
+
+                sk = (WOLFSSL_DIST_POINTS*)XMALLOC(
+                            sizeof(WOLFSSL_DIST_POINTS), NULL,
+                            DYNAMIC_TYPE_ASN1);
+                if (sk == NULL) {
                     return NULL;
                 }
-                obj->type  = CRL_DIST_OID;
-                obj->grp   = oidCertExtType;
-                obj->obj   = x509->CRLInfo;
-                obj->objSz = x509->CRLInfoSz;
-            }
-            else {
+                XMEMSET(sk, 0, sizeof(WOLFSSL_DIST_POINTS));
+                sk->type = STACK_TYPE_DIST_POINT;
+
+                gn = (WOLFSSL_GENERAL_NAME*)XMALLOC(
+                        sizeof(WOLFSSL_GENERAL_NAME), NULL,
+                        DYNAMIC_TYPE_OPENSSL);
+                if (gn == NULL) {
+                    WOLFSSL_MSG("Error creating GENERAL_NAME");
+                    goto err;
```

Comment:
Would prefer not to add more `goto` code to the code base, per Todd's comments at an engineering meeting a couple weeks back.

Reply:
Since this is just expanding on an existing pattern of the function, I think this is fine. We should just not add new functions that use `goto`s.